File: synapse-runtime/common/models/transport/training_start_request.py
from uuid import UUID

from pydantic import BaseModel, Field, ConfigDict

from common.enums.model_status import ModelStatus


class TrainingStartRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    workspace_id: UUID = Field(alias="workspaceId")
    experiment_id: UUID = Field(alias="experimentId")
    status: ModelStatus
    telemetry_topic: str = Field(alias="telemetryTopic")
    publisher_service_url: str = Field(alias="publisherServiceUrl")
    metrics_storage_key: str = Field(alias="metricsStorageKey")

File: synapse-transport/src/main/java/com/synapse/transport/feature/runtime/dto/request/TrainingStartRequest.java
package com.synapse.transport.feature.runtime.dto.request;

import java.util.UUID;
import com.synapse.transport.common.enums.ModelStatus;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TrainingStartRequest {
    private UUID workspaceId;
    private UUID experimentId;
    private ModelStatus status;
    private String telemetryTopic;
    private String publisherServiceUrl;
    private String metricsStorageKey;
}
